lib/segment/src/segment_constructor/segment_builder.rs

```diff
@@ -215,29 +216,41 @@ impl SegmentBuilder {
         Ok(loaded_segment)
     }
 
-    fn update_quantization(segment: &Segment, stopped: &AtomicBool) -> OperationResult<()> {
-        let config = segment.config();
-        for (vector_name, vector_data) in &segment.vector_data {
-            if let Some(quantization) = config.quantization_config(vector_name) {
-                let segment_path = segment.current_path.as_path();
-                check_process_stopped(stopped)?;
+    fn update_quantization(segment: &mut Segment, stopped: &AtomicBool) -> OperationResult<()> {
+        let config = segment.config().clone();
 
-                let vector_storage_path = get_vector_storage_path(segment_path, vector_name);
-                let max_threads = match segment
+        let threads_count: HashMap<String, usize> =
+            HashMap::from_iter(segment.vector_data.keys().map(|vector_name| {
```

Comment:
Is having two passes on the `vector_data` necessary or an optimization?

Reply:
That's a perfect catch! I did this because of the borrow checker. But you are right, it can be simplified. Done!